### PR TITLE
Preventing going back 

### PR DESCRIPTION
**Pull request by AndrewSakovich**

### CODE REVIEW

File: src/components/AddNewItemScreenNavigationHeader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
+import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
+import { COLORS } from '../../COLORS';
+import { useNavigation } from '@react-navigation/native';
+import { AddNewItemScreenNavigationProps } from '../../screens/AddNewItemScreen/type';
+import { createAlertMessageHelper } from '../../helpers/createAlertMessageHelper';
+import { AddNewItemScreenNavigationHeaderPropsType } from './type';
+import { style } from './style';
+
+export const AddNewItemScreenNavigationHeader: FC<
```
Comment (SHEMPEL133) [inline src/components/AddNewItemScreenNavigationHeader/index.tsx]: it's not a header, it's a back button

File: src/screens/AddNewItemScreen/index.tsx [hunk 2]
```diff
@@ -26,6 +27,16 @@ export const AddNewItemScreen: FC = () => {
...
 
+  console.log('screen text', text);
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerLeft: () => {
+        return <AddNewItemScreenNavigationHeader itemText={text} />;
+      },
+    });
+  }, [text, navigation]);
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/index.tsx]: I think it's no good idea to set new button every time when text changes

we should provide function from closure and this function should has access to current text

File: src/components/AddNewItemScreenNavigationHeader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  AddNewItemScreenNavigationHeaderPropsType
+> = props => {
+  const navigation = useNavigation<AddNewItemScreenNavigationProps>();
+  const { itemText } = props;
+  console.log('header text', itemText);
```
Comment (SHEMPEL133) [inline src/components/AddNewItemScreenNavigationHeader/index.tsx]: ?

File: src/components/AddNewItemScreenNavigationHeader/index.tsx (hunk 1 above)
Comment (AndrewSakovich) [reply src/components/AddNewItemScreenNavigationHeader/index.tsx]: ok, i rename files

File: src/screens/AddNewItemScreen/type.ts [hunk 4]
```diff
@@ -1,6 +1,13 @@
-import { RootStackNavigationParamList } from '../../navigators/RootStackNavigator/type';
 import { StackNavigationProp } from '@react-navigation/stack';
 import { NAMESCREEN } from '../../navigators/NAMESCREEN';
+import { NavigatorScreenParams } from '@react-navigation/native';
+import { MainBottomTabParamList } from '../../navigators/MainBottomTabNavigator/type';
+
+export type RootStackNavigationParamList = {
+  [NAMESCREEN.MAIN_BOTTOM_TAB_NAVIGATOR]: NavigatorScreenParams<MainBottomTabParamList>;
+  [NAMESCREEN.ADD_NEW_ITEM_SCREEN]: { isBack: boolean };
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/type.ts]: do we need `isBack`?

Comment (AndrewSakovich) [reply src/screens/AddNewItemScreen/type.ts]: no

File: src/screens/AddNewItemScreen/index.tsx [hunk 5]
```diff
@@ -26,6 +28,32 @@ export const AddNewItemScreen: FC = () => {
...
+    return goBack();
+  };
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerLeft: () => {
+        return <AddNewItemScreenBackButton onPress={onPressBack} />;
+      },
+    });
+  }, [onPressBack]);
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/index.tsx]: your onPressBack is created on every render 
use useCallback (check how times useEffect calls using console.log)

File: src/screens/AddNewItemScreen/index.tsx [hunk 6]
```diff
@@ -26,6 +28,32 @@ export const AddNewItemScreen: FC = () => {
...
+        title: 'Discard changes',
+        cancelButtonTitle: "Don't leave",
+        confirmButtonTitle: 'Leave',
+      });
+    }
+
+    return goBack();
+  };
+
+  useEffect(() => {
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/index.tsx]: you should set option in useLayoutEffect like in the react navigation doc

**Pull request closed without merging**